swarm/swarm.go

```diff
@@ -446,6 +449,15 @@ func (self *Swarm) Stop() error {
 		ch.Stop()
 		ch.Save()
 	}
+	if self.swap != nil {
+		self.swap.Close()
+	}
+	if self.accountingMetrics != nil {
```

Comment:
I feel too much accounting specific stuff on this high level, can we not put this within swap?,
,

Reply:
See above; this specific line is about closing the metrics levelDB and gracefully closing the reporter goroutine